File: Programmers/Lv.1/문자열마음대로정렬하기.py
def solution(strings, n):
    answer = []
    
    for i in range(len(strings)):
        strings[i] = strings[i][n] + strings[i]
    strings.sort()
    
    for j in range(len(strings)):
        answer.append(strings[j][1:])
    
    return answer

# def solution(strings, n):
#     '''strings의 문자열들을 n번째 글자를 기준으로 정렬해서 return하세요'''
#     return sorted(strings, key=lambda x: x[n])

File: Programmers/Lv.1.Kotlin/문자열마음대로정렬하기.kt
class Solution {
    fun solution(strings: Array<String>, n: Int): Array<String> {
        //객체의 속성을 전혀 사용하지 않거나 변경하지 않고 사용하는 경우에 also
        return strings.also {
            it.sort()
            it.sortBy { it[n] }
        }
    }
}